File: sunshineams.py
#display the welcome menu
print("---welcome to Sunshine AMS\n 1.New student\n 2.Registration\n 3.key in student marks\n 4.Print student progressive\n 5.Exit\n press any key from the above numbers--")
grade=()
key=int(input())
if key==1:
    name=input("Enter Full Name\n")
    regNo=input("Enter Registration Number\n")
    course=input("Enter Course Name\n")
    department=input("Enter Department\n")
    print(f"Name:{name}\tRegistration number:{regNo}\tCourse:{course}\tDepartment:{department}")
elif key==2:
    studentname=input("Enter the new student name\n")
    semester=input("Enter the nummber of semesters\n")
    academicyear=input("Enter the current Academic year\n")
    print(f"Name:{studentname}\tSemester:{semester}\tAcademicyear:{academicyear}")
elif key==3:
    print("A student can do a maximum of 8 units and a minimum of 4 in a semester\n")
    unitname=input("Enter Unit Name\n")
    unitcode=input("Enter Unit Code\n")
    sittingcat=int(input("Enter cat mark out of 15\n"))
    assignement=int(input("Enter assignment mark out of 10\n"))
    attendancemark=int(input("Enter attendance marks out of 5\n"))
    exammark=int(input("Enter exam mark out of 70\n"))
    finalmark=sittingcat+assignement+attendancemark+exammark
    if finalmark>=70 and finalmark<=100:
        grade=("A\tremarks\tExcellent")
    if finalmark>=60 and finalmark<=69:
        grade=("B\tremarks\tGood")
    if finalmark>50 and finalmark<=59:
        grade=("C\tremarks\tFair")
    if finalmark>=40 and finalmark<=49:
        grade=("D\tremarks\tPass")
    if finalmark>=0 and finalmark<=39:
        grade=("F\tremarks\tFail")  
    print(f"Unit:{unitname}\tCode:{unitcode}\tCAT:{sittingcat}\tassignment:{assignement}\tattendancemark:{attendancemark}\texammark:{exammark}\tfinalscore:{finalmark}\tgrade:{grade}")
elif key==4:
    registration=input("Enter registartion number\n")
    fullnames=input("Enter full name\n")
    department=input("Enter Department room\n")
    academicyear=input("Enter Academic year 0000/0000 format\n")
    course=input("Enter Course Name\n")
    print("Student Transcript of Sunshine College International")
    print(f"Registration:{registration}\n")
    print(f"Full names:{fullnames}\n")
    print(f"Department:{department}\n")
    print(f"Academicyear:{academicyear}\n")
    print(f"Course:{academicyear}\n")
elif key==5:
    print("Exiting the program")
    exit()
else:
    print("Error")
    exit()

File: C++/sunshineams.cpp
#include<iostream>
#include <iomanip>
using namespace std;
class student
{
    private:
    string name,RegNO,course,department,academic,semester,unit,code,grade,Remarks;
    int key,sitcat,assign,attmark,exam,finalmark;   
    public:
    void accept();
    void display();
};
void student::accept()
{
    cout<<"Welcome to Sunshine AMS"<<endl;
    cout<<"1.New Student\n2.Register Student\n3.Key in Student Marks\n4.Print Student Progressive\n5.Exist"<<endl;
    cout<<"Select any key from the sbove\n";
    cin>>key;
    if (key==1)
    {
        cout<<"Enter Student Name"<<endl;
        cin>>name;
        cout<<"Registration Number"<<endl;
        cin>>RegNO;
        cout<<"Course"<<endl;
        cin>>course;
        cout<<"Department"<<endl;
        cin>>department;
        cout<<"Name"<<setw(10)<<"RegNO"<<setw(20)<<"Course"<<setw(15)<<"Department"<<endl;
        cout<<name<<setw(10)<<RegNO<<setw(20)<<course<<setw(15)<<department<<endl;
    }
    else if(key==2)
    {
        cout<<"Enter the Academic year"<<endl;
        cin>>academic;
        cout<<"Enter Semester"<<endl;
        cin>>semester;
        cout<<"Academic year"<<setw(20)<<"Semester"<<endl;
        cout<<academic<<setw(25)<<semester<<endl;
    }
    else if(key==3)
    {
        cout<<"Unit Name"<<endl;
        cin>>unit;
        cout<<"Unit Code"<<endl;
        cin>>code;
        cout<<"Sitting CAT out of 15"<<endl;
        cin>>sitcat;
        cout<<"Assignment out of 10"<<endl;
        cin>>assign;
        cout<<"Attendance Mark out of 5"<<endl;
        cin>>attmark;
        cout<<"Exam mark out of 70"<<endl;
        cin>>exam;
        finalmark=sitcat+assign+attmark+exam;
        if (finalmark>=70 && finalmark<=100)
        grade='A';
        else if(finalmark>=60 && finalmark<=69)
        grade='B';
        else if(finalmark>=50 && finalmark<=59)
        grade='C';
        else if(finalmark>=40 && finalmark<=49)
        grade='D';
        else if(finalmark>=0 && finalmark<=39)
        grade='F';    
    }
    else if(key==4)
    {
        cout<<"The student progressive results"<<endl; 
        cout<<"Registration Number\nFull Name\nDepartment"<<endl;  
    }
    else if(key==5)
    {
        cout<<"Closing the program"<<endl;        
    }
    else
    {
        cout<<"error"<<endl;
    }
    while (key==1 && key==2 && key==3 && key==4)
    {
        cout<<"Welcome to Sunshine AMS"<<endl;
        cout<<"1.New Student\n2.Register Student\n3.Key in Student Marks\n4.Print Student Progressive\n5.Exist"<<endl;
        
    }
}
void student::display()
{
    cout<<"Unit"<<setw(10)<<"Code"<<setw(10)<<"CAT"<<setw(15)<<"Assignment"<<setw(20)<<"Attendancemark"<<setw(10)<<"Exam"<<setw(10)<<"Final"<<setw(10)<<"Grade"<<endl;
    cout<<unit<<setw(10)<<code<<setw(10)<<sitcat<<setw(10)<<assign<<setw(15)<<attmark<<setw(20)<<exam<<setw(10)<<finalmark<<setw(10)<<grade<<endl;
    
}
int main()
{
    student s;
    s.accept();
    s.display();
    return 0;  
}
